plonky2/src/plonk/circuit_builder.rs

```diff
@@ -669,27 +664,16 @@ impl<F: RichField + Extendable<D>, const D: usize> CircuitBuilder<F, D> {
             "FRI total reduction arity is too large.",
         );
 
-        let gates = self.gates.iter().cloned().collect();
-        let (gate_tree, max_filtered_constraint_degree, num_constants) = Tree::from_gates(gates);
-        let prefixed_gates = PrefixedGate::from_tree(gate_tree);
-
-        // `quotient_degree_factor` has to be between `max_filtered_constraint_degree-1` and `1<<rate_bits`.
-        // We find the value that minimizes `num_partial_product + quotient_degree_factor`.
-        let min_quotient_degree_factor = (max_filtered_constraint_degree - 1).max(2);
-        let max_quotient_degree_factor = self.config.max_quotient_degree_factor.min(1 << rate_bits);
-        let quotient_degree_factor = (min_quotient_degree_factor..=max_quotient_degree_factor)
-            .min_by_key(|&q| num_partial_products(self.config.num_routed_wires, q) + q)
-            .unwrap();
-        debug!("Quotient degree factor set to: {}.", quotient_degree_factor);
+        let quotient_degree_factor = self.config.max_quotient_degree_factor;
+        let mut gates = self.gates.iter().cloned().collect::<Vec<_>>();
+        gates.sort_unstable_by_key(|g| (g.0.degree(), g.0.id()));
```

Comment:
It's a little hard to follow since the reason for sorting by degree has to do with the implementation of `selector_polynomials`. What do you think of just sorting by ID here, and then stable-sorting by degree in `selector_polynomials`? It'd be slightly wasteful but the list should be tiny, and I think it'd be easy to follow the reasoning.

In any case I think we should keep some comment; what we had before was

```rust
        // The HashSet of gates will have a non-deterministic order. When converting to a Vec, we
        // sort by ID to make the ordering deterministic.
```

Reply:
Oh I see, `SelectorsInfo` is storing gate indices so I guess `gates` here needs the same ordering involving degree. Let's just explain why it's needed then.